xmodule/library_content_block.py

```diff
@@ -668,15 +704,23 @@ def source_library_values(self):
 
     def editor_saved(self, user, old_metadata, old_content):  # lint-amnesty, pylint: disable=unused-argument
         """
-        If source_library_id or capa_type has been edited, refresh_children automatically.
+        If source_library_id is empty, clear source_library_version and children.
         """
-        old_source_library_id = old_metadata.get('source_library_id', [])
-        if (old_source_library_id != self.source_library_id or
-                old_metadata.get('capa_type', ANY_CAPA_TYPE_VALUE) != self.capa_type):
-            try:
+        if not self.source_library_id:
+            self.children = []  # lint-amnesty, pylint: disable=attribute-defined-outside-init
+            self.source_library_version = ""
+        else:
+            self.source_library_version = str(self.tools.get_library_version(self.source_library_id))
+
+    def post_editor_saved(self):
+        """
+        If xblock has been edited, refresh_children automatically.
```

Comment:
[Editor_saved is called BEFORE the modulestore is updated with new values. ](https://github.com/openedx/edx-platform/blob/779e4b9e7d3abcd647ffa3b0a6f544788980d2b3/cms/djangoapps/contentstore/xblock_storage_handlers/view_handlers.py#L358)

As refresh_children needs to pull those new values from the modulestore, it has to be called afterward, is my guess. That may be a wrong interpretation of the code, however.




Refresh_children needs to pull from the modulestore. Hence, it can a